Detections/OfficeActivity/MultipleTeamsDeletes.yaml

```diff
@@ -0,0 +1,23 @@
+id: 173f8699-6af5-484a-8b06-8c47ba89b380
+name: Multiple Teams deleted by a single user
+description: |
+  'This detection flags the occurrences of deleting multiple teams within an hour.
+  This data is a part of Office 365 Connector in Azure Sentinel.'
+tactics:
+  - Impact
+relevantTechniques:
+  - T1485
+  - T1489
+query: |
+
+  // Adjust this value to change how many Teams should be deleted before including
+  let max_delete = 3;
+  // Adjust this value to change the timewindow the query runs over
+  let time_window = 1d;
+  OfficeActivity
+  | where OfficeWorkload == "MicrosoftTeams" 
+  | where TimeGenerated > ago(time_window)
+  | where Operation =~ "MemberAdded"
+  | summarize make_set(ItemName) by UserId
+  | where set_ItemName > max_delete
```

Comment:
In order to check if a set or list has a certain count of items, you have to use arraa_length.  In this case, we also need to change to the field set_TeamName which I have cast to DeletedTeams field name for better context - | where array_length(DeletedTeams) > max_delete

Reply:
Updated for count correction.